reframe/frontend/statistics.py

```diff
@@ -197,6 +199,42 @@ def json(self, force=False):
 
         return self._run_data
 
+    def junit(self, json_report, force=False):
+        # https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd
+        xml_testsuites = ET.Element('testsuites')
+        xml_testsuite = ET.SubElement(
+            xml_testsuites, 'testsuite',
+            attrib={
+                'name': 'rfm',
+                'errors': '0',
+                'failures': str(json_report['session_info']['num_failures']),
+                'tests': str(json_report['session_info']['num_cases']),
+                'time': str(json_report['session_info']['time_elapsed']),
+                'hostname': json_report['session_info']['hostname'],
+            }
+        )
+
+        for testid in range(len(json_report['runs'][0]['testcases'])):
+            tid = json_report['runs'][0]['testcases'][testid]
+            name = (
+                f"{tid['name']} on {tid['system']} using {tid['environment']}"
+            )
+            testcase = ET.SubElement(
+                xml_testsuite, 'testcase',
+                attrib={
+                    'classname': tid['filename'],
+                    'name': name,
+                    'time': str(tid['time_total']),
+                }
+            )
+            if not tid['result'] == 'success':
+                testcase_msg = ET.SubElement(
+                    testcase, 'failure', attrib={'message': tid['fail_phase']}
+                )
+                testcase_msg.text = tid['fail_reason']
+
+        return ET.tostring(xml_testsuites, encoding='utf8', method='xml')
```

Comment:
Can't the output be formatted in multiple lines instead of everything in a single line? PS: I'm just judging from the example XML file you've posted in the description of this PR.